include/dr/detail/communicator.hpp

```diff
@@ -43,6 +43,12 @@ class communicator {
     MPI_Scatter(src, size, MPI_BYTE, dst, size, MPI_BYTE, root, mpi_comm_);
   }
 
+  template <typename T>
+  void scatter(const std::vector<T> &src, T &dst, std::size_t root) const {
+    assert(src.size() >= size());
+    scatter(src.data(), &dst, sizeof(T), root);
+  }
+
```

Comment:
```  
  void scatter(std::contiguous_range auto &&src, T &dst, std::size_t root) const {
    assert(rng::size(src) >= size());
    scatter(rng::data(src), &dst, sizeof(T), root);
  }
```

or maybe std::span instead of std::vector

Reply:
I made it consistent with `void gather(const T &src, std::vector<T> &dst, std::size_t root)` where vector is used, but sure, both can be `std::span`. Done.

Reply:
I thought there was an implicit conversion from vector to span, but apparently not.